pkg/codeconfig/codeconfig.go

```diff
@@ -110,21 +123,46 @@ func (s *Stack) GetApiSpec(api string) (*openapi3.T, error) {
 	return doc, nil
 }
 
-func (s *Stack) String() string {
-	funcStrings := make([]string, 0)
-
-	for _, f := range s.functions {
-		funcStrings = append(funcStrings, f.String())
+func (c *CodeConfig) ToStack() *stack.Stack {
+	s := &stack.Stack{
+		Name:        path.Base(c.stackPath),
+		Functions:   map[string]stack.Function{},
+		Collections: map[string]interface{}{},
+		Containers:  map[string]stack.Container{},
+		Buckets:     map[string]interface{}{},
+		Topics:      map[string]interface{}{},
+		Queues:      map[string]interface{}{},
+		Schedules:   map[string]stack.Schedule{},
+		Apis:        map[string]string{},
+		Sites:       map[string]stack.Site{},
+		EntryPoints: map[string]stack.Entrypoint{},
 	}
-
-	return fmt.Sprintf(`
-	  functions: [%s]
-	`, strings.Join(funcStrings, ",\n"))
-}
-
-func NewStack() *Stack {
-	return &Stack{
-		functions: make([]*Function, 0),
-		lock:      sync.RWMutex{},
+	for handler, f := range c.functions {
+		name := strings.Replace(path.Base(handler), path.Ext(handler), "", 1)
+		s.Functions[name] = stack.Function{Handler: handler}
+		for k, v := range f.apis {
+			s.Apis[k] = v.workers[0].String()
+		}
+		for k, v := range f.buckets {
+			s.Buckets[k] = v.String()
+		}
+		for k, v := range f.collections {
+			s.Collections[k] = v.String()
+		}
+		for k, v := range f.queues {
+			s.Queues[k] = v.String()
+		}
+		for k, v := range f.schedules {
+			s.Schedules[k] = stack.Schedule{
+				Expression: v.String(),
+				Target:     stack.ScheduleTarget{},
+				Event:      stack.ScheduleEvent{},
+			}
```

Comment:
We should have enough information to provide a target here.

We do need to setup the event as well, but need to do a little bit of design work so we can identify the callback function to route to internally (need just enough information to identify that target internally).